crypto/cryptlib.c

```diff
@@ -105,7 +105,7 @@ void OPENSSL_cpuid_setup(void)
 }
 #endif
 
-#if defined(_WIN32) && !defined(__CYGWIN__)
+#if defined(_WIN32) && !defined(__CYGWIN__) && !defined(OPENSSL_SYS_WIN_CORE)
```

Comment:
It's not about just pushing through compilation at *any* cost. So this would need closer consideration. As temporary measure one can link even with onecore_downlevel.lib.

Reply:
According to microsoft reference:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms683238(v=vs.85).aspx
https://msdn.microsoft.com/en-us/library/windows/desktop/ms683225(v=vs.85).aspx
https://msdn.microsoft.com/en-us/library/windows/desktop/ms645505(v=vs.85).aspx
Specified in given ifdef, are:

Minimum supported client | Windows 2000 Professional [desktop apps only]

It's not just about passing compilation, it is rather about to avoid functionalities available only in desktop x86 x64 system libraries. There is no User32.dll on some versions of windows.
To be OneCore capable is to link system functions only through onecore dll's.
That bottleneck library set (actually of 4), assure that this API can run at every Windows on every platform, which is main concept of OneCore.

Reply:
I think that the point is that we want `OPENSSL_showfatal` and `OPENSSL_isservice` to work correctly in all circumstances...  so if we look at `OPENSSL_showfatal`, what's the proper onecore way to through a message in the face of the user?  And if we look at `OPENSSL_isservice`, what's the proper oncore way to detect that the running program is an interactive one and not a service of some sort?

Reply:
From what I've observed, there is no MessageBox nor gui functions in OneCore and it seems that OneCore OpenSSL should be treated as no-gui by definition. No-Gui openSSL today uses EventLogger, but this one has been removed from API, Microsoft prepared two modern ways to send events int his case, both are via ETW: 
-ManifestBased: 
https://msdn.microsoft.com/en-us/library/windows/desktop/aa385772(v=vs.85).aspx
There is need to generate manifest file, compile it via MC, and include generated header file. Additionally it needs linking into library additional res file generated via MC, derived from Manifest. 
It looks quiet complicated and requires installing manifest on target machine via wevtutil.exe
After that events can come to specified directory within Viewer or system channels like Application
-TraceLogging:
https://msdn.microsoft.com/en-us/library/windows/desktop/dn904630(v=vs.85).aspx
 It is some kind of newer version of actual solution, by simply including TraceLoggingProvider.h from SDK it is possible to register provider send self-describing (manifestless) event and unregister provider like was done before. The problem is I am currently not able to see these events within EventViewer.exe, even if i provide channel number during implementation, lack of any documentation doesn't help either. Still events can be tracked by other event consumers like TraceView, Logman tho. Wonder if this solution is good enough.